pkg/render/guardian/guardian.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guardian
+
+import (
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	_k8s "github.com/tigera/operator/pkg/k8s"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	GuardianContainerName              = "guardian"
+	GuardianServiceName                = "tigera-guardian"
+	ManagedClusterConnectionSecretName = "tigera-managed-cluster-connection"
+)
+
+func Container(tunnelCAType operatorv1.CAType, voltronURL string,
+	bundle certificatemanagement.TrustedBundle, tunnelSecret *corev1.Secret, proxyEnvVars []corev1.EnvVar) *_k8s.Container {
+
+	return _k8s.NewContainer(GuardianContainerName, components.Component{
```

Comment:
Why the underscore? I don't think thats a pattern we use commonly. Probably should just be `k8s`?

```suggestion
	return k8s.NewContainer(GuardianContainerName, components.Component{
```